spec/requests/saml_sessions_controller_spec.rb

```diff
@@ -61,16 +64,16 @@
           let(:provider) { create(:provider, :created_by_devise, :with_ccms_apply_role, username:) }
 
           it "calls the Provider details api" do
-            expect(ProviderDetailsCreator).to receive(:call).with(provider).and_call_original
+            expect(PDA::ProviderDetailsCreator).to receive(:call).with(provider).and_call_original
             post_request
           end
 
           it "updates the record with firm and offices" do
             post_request
-            firm = Firm.find_by(ccms_id: raw_details_response[:providerFirmId])
+            firm = Firm.find_by(ccms_id: raw_details_response.dig(:firm, :ccmsFirmId))
```

Comment:
My preference for the line below (deleting this line). feel free to ignore. I generally prefer to use literals for comparison. These can be search for/greped easily enough if necessary, and in the case of text values in particular (not these), are more readable and understandable.

```suggestion
  expect(provider.firm_id).to eq 10_835_831
```